### ISSUE TITLE
Windows image doesn't support file length longer than 260 characters even after setting LongPathsEnabled to 1

### ISSUE THREAD

**Issue by Bruceliu-rs** [microsoft/Windows-Containers#361]
Please fill out _all the sections_ below for bug issues, otherwise it'll be closed as it won't be actionable for us to address.

**Describe the bug**
Windows image doesn't support file length longer than 260 characters even after setting LongPathsEnabled to 1.

**To Reproduce** 
Use this image: mcr.microsoft.com/windows/servercore:ltsc2022
use powershell
run the following command to enable long path
New-ItemProperty -Path "HKLM:\SYSTEM\CurrentControlSet\Control\FileSystem" -Name "LongPathsEnabled" -Value 1 -PropertyType DWORD -Force
run the following command to create a file path to make sure the full path is 261 characters.
new-item '1234567890abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890abcdefghijklmnopqrstuvwxyzABCDEFGHI.txt'

for example, the following file has 261 characters, C:\builds\testfolder\citest\1234567890abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890abcdefghijklmnopqrstuvwxyzABCDEF.txt

new-item : Could not find a part of the path 'C:\builds\folder1\citest\12345678
90abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890abcdefghijklmno
pqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890abcdefghijklmnopqrstuvwxyzABCDEF
GHIJKLMNOPQRSTUVWXYZ1234567890abcdefghijklmnopqrstuvwxyzABCDEFGHI.txt'.
At line:1 char:1
+ new-item '1234567890abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVW ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : WriteError: (C:\builds\folde...yzABCDEFGHI.txt:S 
   tring) [New-Item], DirectoryNotFoundException
    + FullyQualifiedErrorId : NewItemIOError,Microsoft.PowerShell.Commands.New 
   ItemCommand

**Expected behavior**
The new file should be created successfully

**Configuration:**
both
mcr.microsoft.com/windows:1809
and
mcr.microsoft.com/windows/servercore:ltsc2022
are failed

**Additional context**
The docker image was created by gitlab-ci, but it was failed when I tried docker windows in my laptop as well.

**Comment by iankingori** [comment]
Hey @Bruceliu-rs,

Most times registry edits require an application restart to take effect(in this case PowerShell).

I've tried building this image:

```
FROM mcr.microsoft.com/windows/servercore:ltsc2022
SHELL ["powershell.exe", "-Command", "$ErrorActionPreference = 'Stop'; $ProgressPreference = 'SilentlyContinue';"]
RUN  New-ItemProperty -Path "HKLM:\SYSTEM\CurrentControlSet\Control\FileSystem" -Name "LongPathsEnabled" -Value 1 -PropertyType DWORD -Force
```

Then ran: `docker.exe run -it <image-name> powershell.exe` and the long path you shared works.
![image](https://github.com/microsoft/Windows-Containers/assets/50120072/87c6161b-76e5-41c6-a9a1-228d67ffa3e2)

Additionally, if the directory structure you provided does not exist, use `New-Item <path> -Force` to create the directories as well. See this for additional info: https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.management/new-item?view=powershell-7.3#example-8-use-the-force-parameter-to-attempt-to-recreate-folders

**Comment by Bruceliu-rs** [comment]
Hi @iankingori, 
    Thanks for the reply. Are there any way to apply the setting without restart? I am using image from another team, so I can't change the Dockerfile.
    Thanks.

**Comment by iankingori** [comment]
@Bruceliu-rs,

This is hacky but works, perhaps there's a better approach:

In the containers' PowerShell terminal:

1. Update the reg key: 
```New-ItemProperty -Path "HKLM:\SYSTEM\CurrentControlSet\Control\FileSystem" -Name "LongPathsEnabled" -Value 1 -PropertyType DWORD -Force```
2. Run the create file command in a new PowerShell instance: 
``` powershell -Command New-Item <long path> -Force```

![image](https://github.com/microsoft/Windows-Containers/assets/50120072/a3e660cc-36f7-4305-871e-2acab9fd3982)

**Comment by Bruceliu-rs** [comment]
@iankingori , Thanks for the sharing. 
I tested it, it works fine, but my use case is with git for windows, what I found is Git can't create file with long file name and long folder names. I guess this workaround may not work with Git for windows.
Btw, I found the real file name limit is 255 characters and the folder's length doesn't matter.
Here is my test:
$mycmd = 'new-item ''dddddddddddddddddddddddddddddddddddddddddddddddddddd
dddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddd
dddddddddddddddddddddddddddddddddddddddddddddddddddddddddd333uuuuuuuuuuuu5557  ffee123.txt'''
PS C:\shared\testtestttt\folder\folder2> powershell -command $mycmd -force
new-item : The specified path, file name, or both are too long. The fully qualified file name must be less than
260 characters, and the directory name must be less than 248 characters.
...

And if I remove one character from the file name, the file can be created successfully.
